tests/integration/src/install_tests.rs

```diff
@@ -0,0 +1,961 @@
+        .expect("failed to get request policies")
+        .0
+        .expect("failed to get request policies");
+
+    if listed_request_policies.policies.len()
+        != expected_request_policies.len() + expected_extra_request_policies
+    {
+        return Err(format!(
+            "expected {} request policies, got {}",
+            expected_request_policies.len() + expected_extra_request_policies,
+            listed_request_policies.policies.len()
+        ));
+    }
+
+    Ok(())
+}
+
+fn assert_initial_named_rules(
+    env: &PocketIc,
+    canister_id: Principal,
+    requester: Principal,
+    expected_named_rules: &Vec<InitNamedRuleInput>,
+) -> Result<(), String> {
+    let listed_named_rules = list_named_rules(env, canister_id, requester)
+        .expect("failed to get named rules")
+        .0
+        .expect("failed to get named rules");
+
+    if expected_named_rules.len() != listed_named_rules.named_rules.len() {
+        return Err(format!(
+            "expected {} named rules, got {}",
+            expected_named_rules.len(),
+            listed_named_rules.named_rules.len()
+        ));
+    }
+
+    for expected_named_rule in expected_named_rules {
+        let _named_rule = listed_named_rules
+            .named_rules
+            .iter()
+            .find(|named_rule| named_rule.name == expected_named_rule.name)
+            .ok_or(format!("named rule {} not found", expected_named_rule.name))?;
+    }
+
+    Ok(())
+}
+
+fn assert_initial_assets(
+    env: &PocketIc,
+    canister_id: Principal,
+    requester: Principal,
+    expected_assets: &Vec<InitAssetInput>,
+) -> Result<(), String> {
+    let listed_assets = list_assets(env, canister_id, requester)
+        .expect("failed to call list_assets")
+        .0
+        .expect("failed to list assets");
+
+    if expected_assets.len() != listed_assets.assets.len() {
+        return Err(format!(
+            "expected {} assets, got {}",
+            expected_assets.len(),
+            listed_assets.assets.len()
+        ));
+    }
+
+    for expected_asset in expected_assets {
+        let asset = listed_assets
+            .assets
+            .iter()
+            .find(|asset| asset.id == expected_asset.id)
+            .ok_or(format!("asset {} not found", expected_asset.id))?;
+
+        if asset.id != expected_asset.id
+            || asset.name != expected_asset.name
+            || asset.blockchain != expected_asset.blockchain
+            || asset.standards != expected_asset.standards
+            || asset.metadata != expected_asset.metadata
+            || asset.symbol != expected_asset.symbol
+            || asset.decimals != expected_asset.decimals
+        {
+            return Err(format!("asset {} does not match expected asset", asset.id));
+        }
+    }
+
+    Ok(())
+}
+
+fn compare_arrays<T: PartialEq>(a: &[T], b: &[T]) -> bool {
+    a.len() == b.len() && a.iter().all(|item| b.contains(item))
```

Comment:
```suggestion
    a.len() == b.len() && a.iter().all(|item| b.contains(item)) && b.iter().all(|item| a.contains(item))
```
to cover the case of duplicate entries